ntoskrnl/po/power.c

```diff
@@ -480,18 +574,36 @@ PoSetHiberRange(IN PVOID HiberContext,
 /*
  * @implemented
  */
+_IRQL_requires_max_(DISPATCH_LEVEL)
+_IRQL_requires_same_
 NTSTATUS
 NTAPI
-PoCallDriver(IN PDEVICE_OBJECT DeviceObject,
-             IN OUT PIRP Irp)
+PoCallDriver(
+    _In_ PDEVICE_OBJECT DeviceObject,
+    _In_ PIRP Irp)
 {
-    NTSTATUS Status;
+    PIO_STACK_LOCATION NextStack;
 
-    /* Forward to Io -- FIXME! */
-    Status = IoCallDriver(DeviceObject, Irp);
+    ASSERT(KeGetCurrentIrql() <= DISPATCH_LEVEL);
 
-    /* Return status */
-    return Status;
+    ASSERT(DeviceObject);
+    ASSERT(Irp);
+
+    NextStack = IoGetNextIrpStackLocation(Irp);
+    ASSERT(NextStack->MajorFunction == IRP_MJ_POWER);
+
+    /* Set DeviceObject for PopPresentIrp */
+    NextStack->DeviceObject = DeviceObject;
+
```

Comment:
Windows checks DeviceObject->Flags here.
``` c
    if (DeviceObject->Flags & 0x8000)
    {
        ...
        return STATUS_SUCCESS;
    }

```
Most likely it is DO_POWER_NOOP. (https://www-user.tu-chemnitz.de/~heha/oney_wdm/ch08d.htm)

Just in case, you can simply add something:
``` c
    if (DeviceObject->Flags & 0x8000)
    {
        DPRINT1("PoCallDriver(%p, %p). Flags - %08X\n", DeviceObject, Irp, DeviceObject->Flags);
    }

```